src/service/db/alerts/alert.rs

```diff
@@ -293,3 +274,130 @@ pub async fn reset() -> Result<(), anyhow::Error> {
     let key = "/alerts/";
     Ok(db::delete(key, true, db::NO_NEED_WATCH, None).await?)
 }
+
+/// Returns the key used to store alerts in the in-memory cache, grouped by
+/// stream.
+fn cache_stream_key(org: &str, stream_type: StreamType, stream_name: &str) -> String {
+    format!("{org}/{stream_type}/{stream_name}")
+}
+
+/// Returns the key used to schedule a trigger for the alert.
+fn scheduler_key(stream_type: StreamType, stream_name: &str, alert_name: &str) -> String {
+    format!("{stream_type}/{stream_name}/{alert_name}")
+}
+
+/// Helper functions for sending events to cache watchers in the cluster.
+mod cluster {
```

Comment:
nice! we can probably further refactor this module out to be useful for watching other tables.